fms/datasets/text.py

```diff
@@ -0,0 +1,79 @@
+import torch
+from torch.utils.data import Dataset
+import requests
+from fms.utils import tokenizers
+
+
+class FromString(Dataset):
```

Comment:
Not sure if `FromString` name captures everything being done here, maybe something that captures the sliding operation it is doing to get the input and labels. This looks to be basically doing some form of sliding window with step_size=1.

Reply:
maybe `CausalTextDatasetFromString`?

I think all causal language modeling datasets are sliding window like this.

Reply:
Yes, i think this name is more descriptive